src/reporter/timex.py

```diff
@@ -84,23 +65,27 @@ def select_time_index_value(headers: dict, notification: dict) -> datetime:
     :param notification: the notification JSON payload as received from Orion.
     :return: the value to be used as time index.
     """
-    custom_index = to_datetime(
-                            _custom_time_index_attribute(headers, notification))
-    time_instant = to_datetime(_time_instant_attribute(notification))
-    meta_time_instant = latest_from_str_rep(
-                                _iter_time_instant_in_metadata(notification))
-    date_modified = to_datetime(_date_modified_attribute(notification))
-    meta_date_modified = latest_from_str_rep(
-                                _iter_date_modified_in_metadata(notification))
-    default_value = datetime.now()
-
-    priority_list = [
-        custom_index, time_instant, meta_time_instant, date_modified,
-        meta_date_modified, default_value
+    considered_attributes = [
+        maybe_value(headers, TIME_INDEX_HEADER_NAME),
+        "TimeInstant",
+        "timestamp",
+        "dateModified"
     ]
 
-    return [d for d in priority_list if d][0]
```

Comment:
Um, why trade a simple algorithm and self-documenting code for the apparently more complicated loop down below? Isn't it possible to just add another entry to the priority list for "timestamp"? 
Just trying to understand the reasoning behind it, since I've most likely missed something... :-)

Reply:
I changed the priority list algorithm because there is no need to calculate all the entries for the priority list and then only return the first item that has a value. Instead, it should be more efficient to calculate the entries one by one and return immediately when a proper value is found.

I made the loop because I didn't want to copy paste the same thing 4 times. I.e. the custom attribute, <code>TimeInstant</code>, <code>timestamp</code> and <code>dateModified</code> are all handled in the same way.

Reply:
Yah, I see your point, thanks for clarifying. But in terms of efficiency, quite frankly I don't think it's going to make a difference in practical terms. If you try running e.g.

    python -m pytest -vv --durations=0 reporter/tests/test_timex.py

you'll see the reported duration of each test is 0.00s---I bet you we're talking microseconds here, not even millis! So I'd rather keep the earlier approach since it seems cleaner? Also both approaches require linear time to get the job done---well assuming dictionary lookups are `O(1)` which they should I think. So in practice I reckon you'd need a JSON payload with hundreds of attributes to appreciate any difference in performance. On another note, a quick way to keep the old code and the performance improvement you suggest would be to make `priority_list` lazy---i.e. using generators, as in `custom_index = yield to_datetime(...)`.

Anyhoo, if you'd rather keep your code the way it is, I'm fine with that too :-)

Reply:
I changed the algorithm to be more explicit and hopefully more readable.

Reply:
@heikkilv wow! awesome sauce! thank you sooo much for this, much appreciated!!!

Reply:
lovely